istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,714 @@
+
+	return cmd
+}
+
+func newEnableCmd() *cobra.Command {
+	opts := &enableCliOptions{}
+	cmd := &cobra.Command{
+		Use:   "enable",
+		Short: "Enable webhook configurations",
+		Example: `
+# Enable the webhook configuration of Galley with the given webhook configuration
+istioctl experimental post-install webhook enable --validation --validation-secret istio.webhook.galley 
+    --namespace istio-system --validation-path validatingwebhookconfiguration.yaml
+
+# Enable the webhook configuration of Galley with the given webhook configuration and CA certificate
+istioctl experimental post-install webhook enable --validation --validation-secret istio.webhook.galley 
+    --namespace istio-system --validation-path validatingwebhookconfiguration.yaml --ca-bundle-file ./k8s-ca-cert.pem
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// --namespace is used as the namespace of the webhook secret
+			opts.webhookSecretNameSpace = namespace
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				return fmt.Errorf("err when creating Kubernetes client interface: %v", err)
+			}
+			err = enableWebhookConfig(client, opts)
+			if err != nil {
+				return fmt.Errorf("err when enabling webhook configurations: %v", err)
+			}
+			fmt.Println("webhook configurations have been enabled")
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.enableValidationWebhook, "validation", true, "Enable "+
+		"validatation webhook (default true).")
+	flags.BoolVar(&opts.enableMutationWebhook, "injection", true, "Enable "+
+		"injection webhook (default true).")
+	// Specifies the local file path to webhook CA bundle.
+	// If empty, will read CA bundle from default Kubernetes service account.
+	flags.StringVar(&opts.caCertPath, "ca-bundle-file", "",
+		"PEM encoded CA bundle which will be used to validate the webhook's server certificates. "+
+			"If this is empty, the kube-apisever's root CA is used if it can be confirmed to have signed "+
+			"the webhook's certificates. This condition is sometimes true but is not guaranteed "+
+			"(see https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping)")
+	flags.StringVar(&opts.validationWebhookConfigPath, "validation-path", "",
+		"The file path of the validation webhook configuration.")
+	flags.StringVar(&opts.mutatingWebhookConfigPath, "injection-path", "",
+		"The file path of the injection webhook configuration.")
+	flags.StringVar(&opts.validatingWebhookServiceName, "validation-service", "istio-galley",
+		"The service name of the validation webhook to manage.")
+	flags.StringVar(&opts.mutatingWebhookServiceName, "injection-service", "istio-sidecar-injector",
+		"The service name of the injection webhook to manage.")
+	flags.StringVar(&opts.webhookSecretName, "webhook-secret", "",
+		"The name of the secret of a webhook. istioctl will verify that the webhook certificate "+
```

Comment:
Is this an existing Kubernetes secret?  If so, please say it.  Your example shows `--webhook-secret istio.webhook.galley` but that fails.

Reply:
Done.